.circleci/config.yml

```diff
@@ -44,12 +44,28 @@ jobs:
             for demo in !($IGNORED_DIRS); do
               if [ -d "$demo" ]; then
                 (
+                  # Call pyenv if we are using MacOS
+                  os="$(uname -s)"
+                  if [ "$os" == "Darwin" ]; then
+                    eval "$(pyenv init -)"
+                    pyenv local $PYTHON
+                  fi
```

Comment:
Since `pyenv local` sets default python across shells, it would be nicer to move (and simplify) this as a separate build step on OSX only. Like we do it in other packages, basically.

Reply:
I don't quite understand what you mean by "`pyenv local` sets default python across shells". 

I am interpreting that statement as it would go across the different circleci `run: `, which is not true. We are calling `eval "$(pyenv init -)"; pyenv local $PYTHON` in `name: Create virtualenv` already, but this command doesn't seem to apply to `run: *install-dependences-template`. Hence the reason why I had the above if-statement in `name: Install dependences`.

I could give macOS its own version of `name: Install dependences`, but I was worried about macOS and linux code diverging too much.

Anyway, please let me know if I've misinterpreted your suggestion.

Reply:
Check [`pyenv local` docs](https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-local).

Reply:
Note to future @m3ller: `pyenv local` will change my local python so long as I stay in my current directory. If I move out of the directory, then I will lose the change.

Current solution is to just grab the correct python version to install virtualenv. Call on virtualenv to create the environment that will contain the desired python version.